glide-core/src/client/reconnecting_connection.rs

```diff
@@ -57,22 +59,59 @@ impl fmt::Debug for ReconnectingConnection {
 
 async fn get_multiplexed_connection(
     client: &redis::Client,
-    push_sender: Option<mpsc::UnboundedSender<PushInfo>>,
+    connection_options: &GlideConnectionOptions,
 ) -> RedisResult<MultiplexedConnection> {
     run_with_timeout(
         Some(DEFAULT_CONNECTION_ATTEMPT_TIMEOUT),
-        client.get_multiplexed_async_connection(push_sender),
+        client.get_multiplexed_async_connection(connection_options.clone()),
     )
     .await
 }
 
+#[derive(Clone)]
+struct TokioDisconnectNotifier {
+    disconnect_notifier: Arc<Notify>,
+}
+
+#[async_trait]
+impl DisconnectNotifier for TokioDisconnectNotifier {
+    fn notify_disconnect(&mut self) {
+        self.disconnect_notifier.notify_one();
+    }
+
+    async fn wait_for_disconnect_with_timeout(&self, max_wait: &Duration) {
+        let _ = timeout(*max_wait, async {
+            self.disconnect_notifier.notified().await;
+        })
+        .await;
+    }
+
+    fn clone_box(&self) -> Box<dyn DisconnectNotifier> {
+        Box::new(self.clone())
+    }
+}
+
+impl TokioDisconnectNotifier {
+    fn new() -> TokioDisconnectNotifier {
+        TokioDisconnectNotifier {
+            disconnect_notifier: Arc::new(Notify::new()),
+        }
+    }
+}
+
 async fn create_connection(
     connection_backend: ConnectionBackend,
     retry_strategy: RetryStrategy,
     push_sender: Option<mpsc::UnboundedSender<PushInfo>>,
 ) -> Result<ReconnectingConnection, (ReconnectingConnection, RedisError)> {
     let client = &connection_backend.connection_info;
-    let action = || get_multiplexed_connection(client, push_sender.clone());
+    let connection_options = GlideConnectionOptions {
```

Comment:
It will be cleaner to implement a `with_push_sender` method:
```rust
pub fn GlideConnectionOptions::with_push_sender(push_sender: Option<mpsc::UnboundedSender<PushInfo>>) -> Self {
...
}
```

Instead of this code block

Reply:
There is only one place that does it, so it would be an overkill as of now, in addition the other method for setting the notifier is also missing from the redis-rs